File: src/test/java/no/novari/flyt/history/repository/utils/instanceflowsummary/FilterPropertyTestCase.java
package no.novari.flyt.history.repository.utils.instanceflowsummary;

import lombok.AllArgsConstructor;
import lombok.Getter;
import no.novari.flyt.history.repository.filters.InstanceFlowSummariesQueryFilter;

import java.util.Set;
import java.util.function.Consumer;

@Getter
@AllArgsConstructor
public class FilterPropertyTestCase {
    private final Consumer<InstanceFlowSummariesQueryFilter.InstanceFlowSummariesQueryFilterBuilder> filterModifier;
    private final Set<EventEntitiesAndExpectedSummary> expectedSummaries;
}

File: src/test/kotlin/no/novari/flyt/history/repository/utils/instanceflowsummary/FilterPropertyTestCase.kt
package no.novari.flyt.history.repository.utils.instanceflowsummary

import no.novari.flyt.history.repository.filters.InstanceFlowSummariesQueryFilter

class FilterPropertyTestCase(
    val filterModifier: (InstanceFlowSummariesQueryFilter.InstanceFlowSummariesQueryFilterBuilder) -> Unit,
    val expectedSummaries: Set<EventEntitiesAndExpectedSummary>,
)
